py/server/tests/test_table.py

```diff
@@ -760,13 +762,39 @@ def no_param_func():
         with self.subTest("arguments must be columns or special variables"):
             with self.assertRaises(DHError) as cm:
                 t1 = t.update("X = vectorized_func(99, i)")
-            self.assertIn("can only be columns", str(cm.exception))
+            self.assertIn("can only be distinct columns", str(cm.exception))
 
         with self.subTest("must have at least one parameter"):
             with self.assertRaises(DHError) as cm:
                 t1 = t.update("X = no_param_func()")
             self.assertIn("at least one parameter", str(cm.exception))
 
+        with self.subTest("same column passed in twice"):
+            with self.assertRaises(DHError) as cm:
+                t1 = t.update("X = vectorized_func(i, i)")
+            self.assertIn("multiple times", str(cm.exception))
+
+    def test_vectorization_perf(self):
+
+        def py_plus(p1, p2) -> int:
+            return p1 + p2
+
+        def py_plus2(p1, p2) -> int:
+            return p1 + p2 + 3
+
+        # not auto-vectorizable
+        st = time.perf_counter_ns()
+        t = self.test_table.update("X = py_plus(i, i + 3)")
+        et = time.perf_counter_ns()
+
+        # auto-vectorizable
+        st2 = time.perf_counter_ns()
+        t2 = self.test_table.update("X = py_plus2(i, i)")
```

Comment:
This is again some very weirdness!

In this test, this can be auto vectorized without problems:
```python
def py_plus2(p1, p2) -> int:
    return p1 + p2 + 3
t2 = self.test_table.update("X = py_plus2(i, i)")
```
But in the previous test, this fails during hand vectorization:
```python
@dh_vectorize
 def vectorized_func(p1, p2):
     return p1 + p2
t1 = t.update("X = vectorized_func(i, i)")
```

Reply:
In the 1st case, the function isn't vectorized. It simply falls back to non-vectorized evaluation. In the 2nd case, because we don't support constant arguments yet and the function is already vectorized, we have to fail it.